### PR TITLE
Xapid 630

### PR DESCRIPTION
**Pull request by theganyo**

### CODE REVIEW

File: apigee_sync.go [hunk 1]
```diff
@@ -127,17 +123,23 @@ func pollChangeAgent() error {
 			log.Errorf("Get changes request failed with status code: %d", r.StatusCode)
 			switch r.StatusCode {
 			case http.StatusUnauthorized:
-				token = ""
+				tokenManager.invalidateToken()
```
Comment (sramamoorthy73) [inline apigee_sync.go]: So, we are doing anything fancy by pre-creating valid tokens before hand.

File: token.go [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+		log.Debugf("Got new token: %#v", token)
+
+		if newInstanceID {
+			newInstanceID = false
+			updateApidInstanceInfo()
+		}
+
+		t.token = &token
+		config.Set(configBearerToken, token.AccessToken)
```
Comment (sramamoorthy73) [inline token.go]: The expectation for all the consumers of the plugin is to get from config, as set before?

File: apigee_sync.go (hunk 1 above)
Comment (theganyo) [inline apigee_sync.go]: I am prefetching before expiration, but I'm not testing them in a loop or anything like that.

File: token.go (hunk 2 above)
Comment (theganyo) [inline token.go]: Yes

**Pull request merged**